WEB-INF/classes/gov/noaa/pfel/erddap/handlers/EDDGridLonPM180Handler.java

```diff
@@ -0,0 +1,122 @@
+package gov.noaa.pfel.erddap.handlers;
+
+import com.cohort.array.StringArray;
+import com.cohort.util.String2;
+import gov.noaa.pfel.erddap.dataset.EDD;
+import gov.noaa.pfel.erddap.dataset.EDDGrid;
+import gov.noaa.pfel.erddap.dataset.EDDGridLonPM180;
+import gov.noaa.pfel.erddap.util.EDStatic;
+import org.xml.sax.Attributes;
+import org.xml.sax.SAXException;
+
+public class EDDGridLonPM180Handler extends State {
+    private StringBuilder content = new StringBuilder();
+    private SaxParsingContext context;
+    private State completeState;
+    private String datasetID;
+
+    public EDDGridLonPM180Handler(SaxHandler saxHandler, String datasetID, State completeState, SaxParsingContext context) {
+        super(saxHandler);
+        this.datasetID = datasetID;
+        this.completeState = completeState;
+        this.context = context;
+    }
+
+    public EDDGrid tChildDataset = null;
+    public String tAccessibleTo = null;
+    public String tGraphsAccessibleTo = null;
+    public boolean tAccessibleViaWMS = true;
+    public boolean tAccessibleViaFiles = EDStatic.defaultAccessibleViaFiles;
+    public StringArray tOnChange = new StringArray();
+    public String tFgdcFile = null;
+    public String tIso19115File = null;
+    public String tDefaultDataQuery = null;
+    public String tDefaultGraphQuery = null;
+    public int tReloadEveryNMinutes = Integer.MAX_VALUE;
+    public int tUpdateEveryNMillis = Integer.MAX_VALUE;
+    public int tnThreads = -1;
+    public boolean tDimensionValuesInMemory = true;
+
+    @Override
+    public void startElement(String uri, String localName, String qName, Attributes attributes) throws SAXException {
+        if (localName.equals("dataset")) {
+            if (tChildDataset == null) {
+                String tType = attributes.getValue("type");
+                if (tType == null || !tType.startsWith("EDDGrid")) {
+                    throw new RuntimeException(
+                            "Datasets.xml error: "
+                                    + "The dataset defined in an "
+                                    + "EDDGridLonPM180 must be a subclass of EDDGrid.");
+                }
+
+                String active = attributes.getValue("active");
+                String childDatasetID = attributes.getValue("datasetID");
+                State state = HandlerFactory.getHandlerFor(tType, childDatasetID, active, this, saxHandler, context);
+                saxHandler.setState(state);
+            } else {
+                throw new RuntimeException(
+                        "Datasets.xml error: "
+                                + "The dataset defined in an "
+                                + "EDDGridLonPM180 must be a subclass of EDDGrid.");
+            }
+        } else {
+            String2.log("Unexpected start tag: " + localName);
```

Comment:
If no other start element is expected, why does the endElement have so many tags it supports?

Reply:
Ah. I think we might have a big problem now that I notice.
Since the function was String2.log() I did not notice it till now but on using println() the unexpected tag message shows up. 

And this is a pattern I have used across all the handlers which means that I would either have to go back and add a bunch of empty conditions for the `startElement` function.

Or another option might be to simple get rid of the `String2.log("Unexpected start tag: " + localName);` line from the `default` and `else` conditions.
I believe this is better. Since if there is a unnecessary tag in there it will simple be ignored.
What do you think?

Reply:
It seems that endElement has code supported all (or almost all) of the elements a dataset expects. It might make more sense for the logging of unexpected elements to only be in the endElement function.

Reply:
Yeah I think that's true. Will make that change.